### PR TITLE
feat: Adds Bamboo HR provider config.

### PR DESCRIPTION
**Pull request by Chandu**
Implements #394

### CODE REVIEW

**Comment by Chandu** [comment]
@uncle-tee @bastienbeurier Have added the suggested changes, let me know if any further changes are required.

File: docs/docs/providers/bamboohr.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+```
+
+## Useful links
+
+-   [Web API docs (their REST API)](https://documentation.bamboohr.com/docs/getting-started)
+
+## API specific gotchas
+
+_No gotchas yet, feel free to contribute it (or check out [airtable](airtable.md) for an example)_
```
[the remark this answers is not in the document]
Comment (Chandu) [reply docs/docs/providers/bamboohr.md:43]: The reference to airtable is for example. There are no known gotchas at this point. What is the established norm for this repo in such case? Remove this section? Or keep the template in place ?

[the remark this answers is not in the document]
Comment (Chandu) [reply docs/docs/providers/bamboohr.md:43]: @uncle-tee ^^^

File: packages/webapp/public/images/template-logos/bamboohr.svg
[the remark this answers is not in the document]
Comment (Chandu) [reply packages/webapp/public/images/template-logos/bamboohr.svg:1]: @uncle-tee Have removed the BambooHR logo and pushed the changes

File: docs/docs/providers/bamboohr.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Chandu) [reply docs/docs/providers/bamboohr.md]: @uncle-tee Have removed the `gotchas` section and pushed the changes

**Comment by Chandu** [comment]
@uncle-tee Have applied changes as per the comments and have updated this PR.

File: docs/docs/providers/bamboohr.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+```json
+{
+    "subdomain": "foo"
+}
+```
+
+## Useful links
+
+-   [Web API docs (their REST API)](https://documentation.bamboohr.com/docs/getting-started)
```
[the remark this answers is not in the document]
Comment (Chandu) [reply docs/docs/providers/bamboohr.md:39]: @bastienbeurier  Have added the section with links for "How to register an Application" and "How to register an Application". For the "List of OAuth scopes", there is no seperate link/page dedicated for scopes on BambooHR doc. It just mentions the scope as `openid, email`. I have already mentioned this under `Creating integration and connection` section.

[the remark this answers is not in the document]
Comment (bastienbeurier) [reply docs/docs/providers/bamboohr.md:39]: Sounds good, thank you for making the changes!

**Review by bastienbeurier — APPROVED** [review]

**Pull request merged**